external-crates/move/move-compiler/src/diagnostics/mod.rs

```diff
@@ -245,6 +245,15 @@ impl Diagnostics {
         }
     }
 
+    pub fn remove(&mut self, index: usize) {
```

Comment:
This scares me to my core. Anything we could do to avoid it would be great. 

Mostly in that it pins us to some implementation details that might not always be there in a parallelized future compiler